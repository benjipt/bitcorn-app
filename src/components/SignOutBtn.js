import React from 'react'

export default function SignOutBtn({ handleLogout }) {
    return (
        <div>
            <button onClick={ handleLogout } className="btn btn-outline-dark">Sign Out</button>
        </div>
    )
}
