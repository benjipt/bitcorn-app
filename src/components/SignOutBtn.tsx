interface SignOutBtnProps {
  handleLogout: () => void;
}

export default function SignOutBtn({ handleLogout }: SignOutBtnProps) {
  return (
    <div>
      <button
        onClick={handleLogout}
        className='btn btn-outline-dark'
        data-testid='SignOutBtn-1'>
        Sign Out
      </button>
    </div>
  );
}
