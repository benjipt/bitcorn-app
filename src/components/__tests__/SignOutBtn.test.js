import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import SignOutBtn from '../SignOutBtn'

afterEach(() => {
    cleanup()
})

describe('SignOutBtn', () => {
    
    test('should render SignOutButton component', () => {
        render(<SignOutBtn />)
        const SignOutBtnComponent = screen.getByTestId('SignOutBtn-1')
        expect(SignOutBtnComponent).toBeInTheDocument()
    })
    
    test('matches snapshot', () => {
        const tree = renderer.create(<SignOutBtn />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
