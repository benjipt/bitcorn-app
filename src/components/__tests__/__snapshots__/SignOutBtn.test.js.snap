// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="btn btn-outline-dark"
    data-testid="SignOutBtn-1"
  >
    Sign Out
  </button>
</div>
`;
